app/scripts/migrations/126.1.ts

```diff
@@ -0,0 +1,53 @@
+import { hasProperty, isObject } from '@metamask/utils';
+import { cloneDeep } from 'lodash';
+import {
+  PhishingControllerState,
+  ListNames,
+} from '@metamask/phishing-controller';
+
+type VersionedData = {
+  meta: { version: number };
+  data: Record<string, unknown>;
+};
+
+export const version = 126.1;
+
+/**
+ * This migration removes `providerConfig` from the network controller state.
+ *
+ * @param originalVersionedData - Versioned MetaMask extension state, exactly what we persist to dist.
+ * @param originalVersionedData.meta - State metadata.
+ * @param originalVersionedData.meta.version - The current state version.
+ * @param originalVersionedData.data - The persisted MetaMask state, keyed by controller.
+ * @returns Updated versioned MetaMask extension state.
+ */
+export async function migrate(
+  originalVersionedData: VersionedData,
+): Promise<VersionedData> {
+  const versionedData = cloneDeep(originalVersionedData);
+  versionedData.meta.version = version;
+  transformState(versionedData.data);
+  return versionedData;
+}
+
+function transformState(
+  state: Record<string, unknown>,
+): Record<string, unknown> {
+  if (
+    hasProperty(state, 'PhishingController') &&
+    isObject(state.PhishingController) &&
+    hasProperty(state.PhishingController, 'phishingLists') &&
+    Array.isArray(state.PhishingController.phishingLists)
+  ) {
+    const phishingController =
+      state.PhishingController as PhishingControllerState;
```

Comment:
I'd recommend avoiding the use of imported types in migrations, because migrations operate on state that is on a specific schema. Whereas the types are mutable and they change as the schema changes in the future.

Imported types in migrations become a maintenance burden for future controller updates, and worst-case can nudge devs into making "corrections" on old migrations that break them.

In this case, the conditions you used _should_ suffice, but for some reason TypeScript isn't bringing along the information from the `Array.isArray` check. But when I re-arrange it a little bit, it works without a cast to an imported type:

```suggestion
    hasProperty(state.PhishingController, 'phishingLists')
  ) {
    const phishingController =
      state.PhishingController as PhishingControllerState;
    
    if (!Array.isArray(phishingController.phishingLists)) {
      return state;
    }
```

I don't entirely follow why this works, but it's something about how TypeScript keeps track of type information on a per-variable basis. Creating this new variable brought along some but not all of the inferred type information.